playback/src/player.rs

```diff
@@ -375,15 +387,14 @@ impl PlayerInternal {
         }
     }
 
-    fn run_onstart(&self) {
-        if let Some(ref program) = self.config.onstart {
-            run_program(program)
-        }
-    }
-
-    fn run_onstop(&self) {
-        if let Some(ref program) = self.config.onstop {
-            run_program(program)
+    fn send_event(&mut self, event: PlayerEvent) {
+        match self.config.event_sender {
+            Some(ref s) =>
+                match s.send(event.clone()) {
+                    Ok(_) => info!("Sent event {:?} to event listener.", event),
+                    Err(err) => error!("Failed to send event {:?} to listener: {:?}", event, err)
```

Comment:
Error happens if the receiver is dropped, which is not a big problem. Just ignore the error in that case.